config/apply/ccsm/collapse.js

```diff
@@ -90,70 +94,57 @@ export function collapseIntoOne(cards, useHtml=false) {
-    let reportString = diagnosticReports.map(formatEntry).join('\n* ');
-    let procedureString = procedures.map(formatEntry).join('\n* ');
-    let immunizationString = immunizations.map(formatEntry).join('\n* ');
-    let episodeOfCareString = episodeOfCares.map(formatEntry).join('\n* ');
-
-    if (
-      conditions.length > 0 ||
-      observations.length > 0 ||
-      medications.length > 0 ||
-      procedures.length > 0 ||
-      diagnosticReports.length > 0 ||
-      encounters.length > 0 ||
-      immunizations.length > 0 ||
-      episodeOfCares.length > 0
-    ) { markdown = markdown + '\n\n' + '## History' + '\n\n'; }
-
-    if (episodeOfCareString.length > 0) {
+    let markdown = 
+      (
+        diagnosticReports.length > 0 ||
+        conditions.length > 0 ||
+        observations.length > 0 ||
+        procedures.length > 0 ||
+        medications.length > 0 ||
+        encounters.length > 0 ||
+        immunizations.length > 0 ||
+        episodeOfCares.length > 0
+      ) ? '' : '### Patient History\n\nNo relevant patient history';
+    
+    if (episodeOfCares.length > 0) {
       markdown = markdown + '\n\n' +
         '### Pregnancy Episode' + '\n\n' +
-        '* ' + episodeOfCareString + '\n\n';
+        formatEntry(episodeOfCares);
     }
-
-    if (conditionString.length > 0) {
+    if (conditions.length > 0) {
       markdown = markdown + '\n\n' +
         '### Diagnoses' + '\n\n' +
-        '* ' + conditionString + '\n\n';
+        formatEntry(conditions);
     }
-    if (observationString.length > 0) {
+    if (observations.length > 0) {
       markdown = markdown + '\n\n' +
         '### Observations' + '\n\n' +
-        '* ' + observationString + '\n\n';
+        formatEntry(observations);
     }
-    if (medicationString.length > 0) {
+    if (medications.length > 0) {
       markdown = markdown + '\n\n' +
         '### Medications' + '\n\n' +
-        '* ' + medicationString + '\n\n';
+        formatEntry(medications);
     }
-    if (reportString.length > 0) {
-      markdown = markdown + '\n\n' +
-        '### Labs' + '\n\n' +
-        '* ' + reportString + '\n\n';
+    if (diagnosticReports.length > 0) {
+      markdown = markdown  + '\n\n' + 
+        '### Tests & Labs' + '\n\n' +
+        formatEntry(diagnosticReports);
     }
-    if (procedureString.length > 0) {
+    if (procedures.length > 0) {
       markdown = markdown + '\n\n' +
         '### Procedures' + '\n\n' +
-        '* ' + procedureString + '\n\n';
+        formatEntry(procedures);
     }
-    if (immunizationString.length > 0) {
+    if (immunizations.length > 0) {
       markdown = markdown + '\n\n' +
         '### Immunizations' + '\n\n' +
-        '* ' + immunizationString + '\n\n';
+        formatEntry(immunizations);
     }
-
-    let finalDetails = justOneCard[0].detail + '\n\n' + markdown;
+    
+    let finalDetails = justOneCard[0].detail + '\n\n' + markdown + '\n\n' + group;
```

Comment:
**UI Recommendation**: Looking at an example of a screening recommendation (the last screenshot in your email), it seems odd to have "Source: ____" at the very top, and then "Reference" all the way at the bottom, which really is a summarized version (or sometimes the specific section) of what guideline this is coming from, and then right after that the actual link to the source again. I think we should consider grouping reference and source together, and using more specific terminology to distinguish the two.

Reply:
I believe that is how the CDS Hooks sandbox chooses to arrange `source` - it renders it before the `detail` markdown. Epic's BPA Cards show `source` after the `detail`